src/infrastructure/jobs/AbstractInMemoryJob.ts

```diff
@@ -23,11 +24,29 @@ export type LockConfiguration = {
   lockTimeout: number
 }
 
-export abstract class AbstractBackgroundJob {
+export function createTask(logger: CommonLogger, job: AbstractInMemoryJob) {
+  return new AsyncTask(
+    job.jobId,
+    () => {
+      return job.process()
+    },
+    (error) => {
+      logger.error(
+        stdSerializers.err({
+          name: error.name,
+          message: error.message,
+          stack: error.stack,
+        }),
+      )
+    },
+  )
+}
+
+export abstract class AbstractInMemoryJob {
```

Comment:
I am not the best with names definitely, but wondering why not call it is just AbstractCronJob, I mean, at the end this is a cron job

Reply:
Makes sense. It is not always Cron, though, you can also define a human-readable interval, e. g. "every 30 minutes".

What do you think about AbstractPeriodicJob or AbstractRecurringJob?

Reply:
AbstractPeriodicJob it is, then!